src/resources/2015/BarrelShrine.ts

```diff
@@ -0,0 +1,25 @@
+import { availableAmount, runChoice, visitUrl } from "kolmafia";
+import { get } from "../../property";
+import { $items } from "../../template-string";
+
+const BARRELS = $items`little firkin, normal barrel, big tun, weathered barrel, dusty barrel, disintegrating barrel, moist barrel, rotting barrel, mouldering barrel, barnacled barrel`;
+
+export function have() {
+  return get("barrelShrineUnlocked");
+}
+
+export function smashParty() {
+  if (!have()) return;
+
+  const total = BARRELS.map((i) => availableAmount(i)).reduce(
+    (sum, q) => sum + q,
+    0
+  );
+  if (total <= 0) return;
+
+  visitUrl("inv_use.php?pwd&whichitem=8568&choice=1");
+
+  for (let i = 0; i < total / 100; i++) {
+    runChoice(2);
```

Comment:
consider adding an `if (handlingChoice())` in front of this, just to make failure less loudly if it does ever fail.